### PR TITLE
refactor: rename_annotations function and unit test

### PR DESCRIPTION
**Pull request by fangliu117**
Please review the refactored rename_annotations function at your convenience. 
Here is the updated requirements of refactoring this function for your reference.
https://tracker.nci.nih.gov/browse/DSP-174
Thanks!

### CODE REVIEW

**Comment by fangliu117** [comment]
Clarification for what happened here:
I've already made a pull request titled "refactor: rename_annotations function and unit test" and then realized there was an issue because of test_rename_labels.py, and I've subsequently added another commit to remove that file. After that, I have ensured my local branch was up to date, and pushed the changes again to avoid merge conflicts. I am happy to solve it if more issues emerge.

File: src/spac/transformations.py [hunk 1]
```diff
@@ -320,77 +325,47 @@ def rename_labels(adata, src_annotation, dest_annotation, mappings):
...
-    ...     "2": "group_6",
-    ...     # ...
-    ...     "37": "group_5",
-    ... }
-    >>> dest_annotation = "renamed_annotations"
-    >>> adata = rename_annotations(
-    ...     adata, src_annotation, dest_annotation, mappings)
+    layer : str, optional
+        The name of the layer in the AnnData object to check.
+    feature : str, optional
```
Comment (georgezakinih) [inline src/spac/transformations.py:330]: @fangliu117, I don't see the feature argument used in this function.

File: src/spac/transformations.py [hunk 2]
```diff
@@ -320,77 +325,47 @@ def rename_labels(adata, src_annotation, dest_annotation, mappings):
...
         A dictionary mapping the original annotation labels to
         the new labels.
-
-    Returns
-    -------
-    adata : anndata.AnnData
-        The updated Anndata object with the new column containing the
-        renamed labels.
-
-    Examples
```
Comment (georgezakinih) [inline src/spac/transformations.py:330]: @fangliu117 , that was a good example, can you add it back?

File: src/spac/transformations.py (hunk 1 above)
Comment (fangliu117) [reply src/spac/transformations.py]: Yes, you are correct that the feature parameter is not necessary, so I have removed it.

File: src/spac/transformations.py (hunk 2 above)
Comment (fangliu117) [reply src/spac/transformations.py:330]: Yes, it was added to the latest commit [f1f8b37].

**Pull request merged**